octane/evmengine/keeper/abci.go

```diff
@@ -300,11 +300,22 @@ func (k *Keeper) startBuild(ctx context.Context, appHash common.Hash, timestamp
 		FinalizedBlockHash: headHash,
 	}
 
+	intHeight, err := umath.ToInt64(height)
+	if err != nil {
+		return engine.ForkChoiceResponse{}, errors.Wrap(err, "height conversion")
+	}
+
+	callCtx := sdk.UnwrapSDKContext(ctx).WithBlockHeight(intHeight)
+	withdrawals, err := k.EligibleWithdrawals(callCtx) //nolint:contextcheck // no new ctx
```

Comment:
We used `ctx.Height` inside `EligibleWithdrawals`, assuming it would always be correct. However, we've found that this isn't always the case. Instead of still relying on `ctx.Height`, we should explicitly pass the desired height rather than providing it implicitly via a fake context height.

Reply:
fully agree, but I thought you still want it in ctx 👍